diff -uNr -x .git a/qemu-server/debian/control b/qemu-server/debian/control
--- a/qemu-server/debian/control	2022-09-16 06:25:27.919683004 -0400
+++ b/qemu-server/debian/control	2022-09-16 06:37:17.922362062 -0400
@@ -48,8 +48,6 @@
          pve-ha-manager (>= 3.0-9),
          pve-qemu-kvm (>= 3.0.1-62),
          socat,
-         swtpm,
-         swtpm-tools,
          ${misc:Depends},
          ${perl:Depends},
          ${shlibs:Depends},
diff -uNr -x .git a/qemu-server/PVE/QemuServer/PCI.pm b/qemu-server/PVE/QemuServer/PCI.pm
--- a/qemu-server/PVE/QemuServer/PCI.pm	2022-09-16 06:25:27.919683004 -0400
+++ b/qemu-server/PVE/QemuServer/PCI.pm	2022-09-16 06:40:21.447456935 -0400
@@ -155,7 +155,7 @@
 	net5 => { bus => 0, addr => 23 },
 	vga1 => { bus => 0, addr => 24 },
 	vga2 => { bus => 0, addr => 25 },
-	vga3 => { bus => 0, addr => 26 },
+#	vga3 => { bus => 0, addr => 26 }, #use to pcie2-3,for aarch64
 	hostpci2 => { bus => 0, addr => 27 },
 	hostpci3 => { bus => 0, addr => 28 },
 	#addr29 : usb-host (pve-usb.cfg)
@@ -587,7 +603,7 @@
 	optional => 1,
 	type => 'string', format => 'pve-qm-ide',
 	typetext => '<volume>',
-	description => "This is an alias for option -ide2",
+       description => "This is an alias for option -scsi2",
     },
     cpu => {
 	optional => 1,
@@ -2449,7 +2465,7 @@
 	    if ($@) {
 		$handle_error->("vm $vmid - unable to parse value of '$key' - $@");
 	    } else {
-		$key = 'ide2' if $key eq 'cdrom';
+               $key = 'scsi2' if $key eq 'cdrom';
 		my $fmt = $confdesc->{$key}->{format};
 		if ($fmt && $fmt =~ /^pve-qm-(?:ide|scsi|virtio|sata)$/) {
 		    my $v = parse_drive($key, $value);
@@ -2484,8 +2500,8 @@
     delete $conf->{snapstate}; # just to be sure
 
     if ($conf->{cdrom}) {
-	die "option ide2 conflicts with cdrom\n" if $conf->{ide2};
-	$conf->{ide2} = $conf->{cdrom};
+       die "option scsi2 conflicts with cdrom\n" if $conf->{scsi2};
+       $conf->{scsi2} = $conf->{cdrom};
 	delete $conf->{cdrom};
     }
 
@@ -3137,7 +3153,13 @@
 
     push @$devices, "-chardev", "socket,id=tpmchar,path=$paths->{socket}";
     push @$devices, "-tpmdev", "emulator,id=tpmdev,chardev=tpmchar";
-    push @$devices, "-device", "tpm-tis,tpmdev=tpmdev";
+       my $arch = get_host_arch();
+       if ($arch eq 'aarch64'){
+    push @$devices, "-device", "tpm-tis-device,tpmdev=tpmdev";
+       } else {
+       push @$devices, "-device", "tpm-tis,tpmdev=tpmdev";
+       }
+ 
 }
 
 sub start_swtpm {
@@ -3642,6 +3664,10 @@
 	    push @$devices, '-readconfig', '/usr/share/qemu-server/pve-q35.cfg';
 	}
     }
+   
+     if ($arch eq 'aarch64') { 
+	    push @$devices, '-readconfig', '/usr/share/qemu-server/pve-aarch64.cfg';
+    }
 
     if (defined(my $fixups = qemu_created_version_fixups($conf, $forcemachine, $kvmver))) {
 	push @$cmd, $fixups->@*;
@@ -4078,8 +4104,11 @@
 	$machine_type_min =~ s/\+pve\d+$//;
 	$machine_type_min .= "+pve$required_pve_version";
     }
+       if ($arch eq 'aarch64'){
+    push @$machineFlags, "type=${machine_type_min},gic-version=host";
+       }else{
     push @$machineFlags, "type=${machine_type_min}";
-
+       }
     push @$cmd, @$devices;
     push @$cmd, '-rtc', join(',', @$rtcFlags) if scalar(@$rtcFlags);
     push @$cmd, '-machine', join(',', @$machineFlags) if scalar(@$machineFlags);
diff -ur a/qemu-server/Makefile b/qemu-server/Makefile
--- a/qemu-server/Makefile	2022-09-20 13:48:18.689880380 +0000
+++ b/qemu-server/Makefile	2022-09-20 13:53:42.481337268 +0000
@@ -98,10 +98,10 @@
 	cd ${BUILDDIR}; dpkg-buildpackage -S -us -uc -d
 	lintian ${DSC}
 
-.PHONY: test
-test:
-	PVE_GENERATING_DOCS=1 perl -I. ./qm verifyapi
-	$(MAKE) -C test
+#.PHONY: test
+#test:
+#	PVE_GENERATING_DOCS=1 perl -I. ./qm verifyapi
+#	$(MAKE) -C test
 
 .PHONY: upload
 upload: ${DEB}
diff -ur a/qemu-server/qemu-configs/Makefile b/qemu-server/qemu-configs/Makefile
--- a/qemu-server/qemu-configs/Makefile	2022-09-20 13:48:18.689880380 +0000
+++ b/qemu-server/qemu-configs/Makefile	2022-09-20 13:54:42.076734811 +0000
@@ -2,11 +2,12 @@
 USRSHAREDIR=$(DESTDIR)/usr/share/qemu-server
 
 .PHONY: install
-install: pve-usb.cfg pve-q35.cfg pve-q35-4.0.cfg
+install: pve-usb.cfg pve-q35.cfg pve-q35-4.0.cfg pve-aarch64.cfg
 	install -d $(USRSHAREDIR)
 	install -m 0644 pve-usb.cfg $(USRSHAREDIR)
 	install -m 0644 pve-q35.cfg $(USRSHAREDIR)
 	install -m 0644 pve-q35-4.0.cfg $(USRSHAREDIR)
+	install -m 0644 pve-aarch64.cfg $(USRSHAREDIR)
 
 .PHONY: clean
 clean:
diff -uNr a/qemu-server/PVE/QemuServer/PCI.pm b/qemu-server/PVE/QemuServer/PCI.pm
--- a/qemu-server/PVE/QemuServer/PCI.pm	2022-09-20 11:20:26.315674253 -0400
+++ b/qemu-server/PVE/QemuServer/PCI.pm	2022-09-20 11:21:35.531737036 -0400
@@ -139,20 +139,17 @@
 	ahci0 => { bus => 0, addr => 7 },
 	qga0 => { bus => 0, addr => 8 },
 	spice => { bus => 0, addr => 9 },
-	virtio0 => { bus => 0, addr => 10 },
-	virtio1 => { bus => 0, addr => 11 },
-	virtio2 => { bus => 0, addr => 12 },
-	virtio3 => { bus => 0, addr => 13 },
-	virtio4 => { bus => 0, addr => 14 },
-	virtio5 => { bus => 0, addr => 15 },
-	hostpci0 => { bus => 0, addr => 16 },
-	hostpci1 => { bus => 0, addr => 17 },
-	net0 => { bus => 0, addr => 18 },
-	net1 => { bus => 0, addr => 19 },
-	net2 => { bus => 0, addr => 20 },
-	net3 => { bus => 0, addr => 21 },
-	net4 => { bus => 0, addr => 22 },
-	net5 => { bus => 0, addr => 23 },
+#   pcie.1 => 	{ bus => 0, addr => 10 }, 
+	net0 => { bus => 0, addr => 11 },
+	hostpci4 => { bus => 0, addr => 12 },
+	hostpci5 => { bus => 0, addr => 13 },
+	hostpci6 => { bus => 0, addr => 14 },
+	hostpci7 => { bus => 0, addr => 15 },
+	hostpci8 => { bus => 0, addr => 16 },
+	hostpci19 => { bus => 0, addr => 17 },
+	hostpci10 => { bus => 0, addr => 18 },
+	hostpci0 => { bus => 0, addr => 22 },
+	hostpci1 => { bus => 0, addr => 23 },
 	vga1 => { bus => 0, addr => 24 },
 	vga2 => { bus => 0, addr => 25 },
 	vga3 => { bus => 0, addr => 26 },
@@ -161,60 +158,54 @@
 	#addr29 : usb-host (pve-usb.cfg)
 	'pci.1' => { bus => 0, addr => 30 },
 	'pci.2' => { bus => 0, addr => 31 },
-	'net6' => { bus => 1, addr => 1 },
-	'net7' => { bus => 1, addr => 2 },
-	'net8' => { bus => 1, addr => 3 },
-	'net9' => { bus => 1, addr => 4 },
-	'net10' => { bus => 1, addr => 5 },
-	'net11' => { bus => 1, addr => 6 },
-	'net12' => { bus => 1, addr => 7 },
-	'net13' => { bus => 1, addr => 8 },
-	'net14' => { bus => 1, addr => 9 },
-	'net15' => { bus => 1, addr => 10 },
-	'net16' => { bus => 1, addr => 11 },
-	'net17' => { bus => 1, addr => 12 },
-	'net18' => { bus => 1, addr => 13 },
-	'net19' => { bus => 1, addr => 14 },
-	'net20' => { bus => 1, addr => 15 },
-	'net21' => { bus => 1, addr => 16 },
-	'net22' => { bus => 1, addr => 17 },
-	'net23' => { bus => 1, addr => 18 },
-	'net24' => { bus => 1, addr => 19 },
-	'net25' => { bus => 1, addr => 20 },
-	'net26' => { bus => 1, addr => 21 },
-	'net27' => { bus => 1, addr => 22 },
-	'net28' => { bus => 1, addr => 23 },
-	'net29' => { bus => 1, addr => 24 },
-	'net30' => { bus => 1, addr => 25 },
-	'net31' => { bus => 1, addr => 26 },
+	#'net0' => { bus => 1, addr => 1 },
+	'net1' => { bus => 1, addr => 2 },
+	'net2' => { bus => 1, addr => 3 },
+	'net3' => { bus => 1, addr => 4 },
+	'net4' => { bus => 1, addr => 5 },
+	'net5' => { bus => 1, addr => 6 },
+	'net6' => { bus => 1, addr => 7 },
+	'net7' => { bus => 1, addr => 8 },
+	'net8' => { bus => 1, addr => 9 },
+	'net9' => { bus => 1, addr => 10 },
+	'net10' => { bus => 1, addr => 11 },
+	'net11' => { bus => 1, addr => 12 },
+	'net12' => { bus => 1, addr => 13 },
+	'net13' => { bus => 1, addr => 14 },
+	'net14' => { bus => 1, addr => 15 },
+	'net15' => { bus => 1, addr => 16 },
+	'net16' => { bus => 1, addr => 17 },
+	'net17' => { bus => 1, addr => 18 },
+	'net18' => { bus => 1, addr => 19 },
+	'net19' => { bus => 1, addr => 20 },
+	'net20' => { bus => 1, addr => 21 },
+	'net21' => { bus => 1, addr => 22 },
+	'net22' => { bus => 1, addr => 23 },
+	'net23' => { bus => 1, addr => 24 },
+	'net24' => { bus => 1, addr => 25 },
 	'xhci' => { bus => 1, addr => 27 },
 	'pci.4' => { bus => 1, addr => 28 },
 	'rng0' => { bus => 1, addr => 29 },
 	'pci.2-igd' => { bus => 1, addr => 30 }, # replaces pci.2 in case a legacy IGD device is passed through
-	'virtio6' => { bus => 2, addr => 1 },
-	'virtio7' => { bus => 2, addr => 2 },
-	'virtio8' => { bus => 2, addr => 3 },
-	'virtio9' => { bus => 2, addr => 4 },
-	'virtio10' => { bus => 2, addr => 5 },
-	'virtio11' => { bus => 2, addr => 6 },
-	'virtio12' => { bus => 2, addr => 7 },
-	'virtio13' => { bus => 2, addr => 8 },
-	'virtio14' => { bus => 2, addr => 9 },
-	'virtio15' => { bus => 2, addr => 10 },
-	'ivshmem' => { bus => 2, addr => 11 },
-	'audio0' => { bus => 2, addr => 12 },
-	hostpci4 => { bus => 2, addr => 13 },
-	hostpci5 => { bus => 2, addr => 14 },
-	hostpci6 => { bus => 2, addr => 15 },
-	hostpci7 => { bus => 2, addr => 16 },
-	hostpci8 => { bus => 2, addr => 17 },
-	hostpci9 => { bus => 2, addr => 18 },
-	hostpci10 => { bus => 2, addr => 19 },
-	hostpci11 => { bus => 2, addr => 20 },
-	hostpci12 => { bus => 2, addr => 21 },
-	hostpci13 => { bus => 2, addr => 22 },
-	hostpci14 => { bus => 2, addr => 23 },
-	hostpci15 => { bus => 2, addr => 24 },
+   'virtio0' => { bus => 2, addr => 1 },
+	'virtio1' => { bus => 2, addr => 2 },
+	'virtio2' => { bus => 2, addr => 3 },
+	'virtio3' => { bus => 2, addr => 4 },
+	'virtio4' => { bus => 2, addr => 5 },
+	'virtio5' => { bus => 2, addr => 6 },
+	'virtio6' => { bus => 2, addr => 7 },
+	'virtio7' => { bus => 2, addr => 8 },
+	'virtio8' => { bus => 2, addr => 9 },
+	'virtio9' => { bus => 2, addr => 10 },
+	'virtio10' => { bus => 2, addr => 11 },
+	'virtio11' => { bus => 2, addr => 12 },
+	'virtio12' => { bus => 2, addr => 13 },
+	'virtio13' => { bus => 2, addr => 14 },
+	'virtio14' => { bus => 2, addr => 15 },
+	'virtio15' => { bus => 2, addr => 16 },
+	'ivshmem' => { bus => 2, addr => 17 },
+	'audio0' => { bus => 2, addr => 18 },
+	
 	'virtioscsi0' => { bus => 3, addr => 1 },
 	'virtioscsi1' => { bus => 3, addr => 2 },
 	'virtioscsi2' => { bus => 3, addr => 3 },
diff -uNr a/qemu-server/qemu-configs/pve-aarch64.cfg b/qemu-server/qemu-configs/pve-aarch64.cfg
--- a/qemu-server/qemu-configs/pve-aarch64.cfg	1969-12-31 19:00:00.000000000 -0500
+++ b/qemu-server/qemu-configs/pve-aarch64.cfg	2022-09-20 11:21:01.483703287 -0400
@@ -0,0 +1,30 @@
+[device "pcie.1"]
+  driver = "pci-bridge"
+  bus = "pcie.0"
+  addr = "0xa"
+  chassis_nr = "1"
+
+#[device "pciroot"]
+# driver = "pci-bridge"
+#  bus = "pci.1"
+#  addr = "1.0"
+#  chassis_nr = "1"
+
+[device "pcie.2"]
+  driver = "pci-bridge"
+  bus = "pci.1"
+  addr = "1.0"
+  chassis_nr = "2"
+
+
+[device "pcie.3"]
+  driver = "pci-bridge"
+  bus = "pci.2"
+  addr = "1.0"
+  chassis_nr = "3"
+
+[device "pcie.4"]
+  driver = "pci-bridge"
+  bus = "pci.2"
+  addr = "2.0"
+  chassis_nr = "4"
