diff -u1 -r pve-old/pve-manager/Makefile pve-maker/pve-manager/Makefile
--- pve-old/pve-manager/Makefile	2022-10-10 08:23:17.534027037 +0000
+++ pve-maker/pve-manager/Makefile	2022-10-10 08:29:29.067372843 +0000
@@ -21,8 +21,2 @@
 
-.PHONY: check
-check: bin test www
-	${MAKE} -C bin check
-	${MAKE} -C test check
-	${MAKE} -C www check
-
 .PHONY: dinstall
diff -u1 -r pve-old/pve-manager/debian/changelog pve-maker/pve-manager/debian/changelog
--- pve-old/pve-manager/debian/changelog	2022-10-10 08:23:17.622027408 +0000
+++ pve-maker/pve-manager/debian/changelog	2022-10-10 08:28:42.187372509 +0000
@@ -2,2 +2,12 @@
 
+  * add default cpu type to host
+  
+  * add ramfb support
+
+  * set default cdrom to scsi2
+
+ -- foxi <jiangcuo@bingsin.com>  Mon, 10 Oct 2022 16:00:20 +0200
+
+pve-manager (7.2-11) bullseye; urgency=medium
+
   * d/control: recommend proxmox-offline-mirror-helper
diff -u1 -r pve-old/pve-manager/debian/control pve-maker/pve-manager/debian/control
--- pve-old/pve-manager/debian/control	2022-10-10 08:24:15.066268657 +0000
+++ pve-maker/pve-manager/debian/control	2022-10-10 08:36:56.911562195 +0000
@@ -84,3 +84,3 @@
          pve-xtermjs (>= 4.7.0-1),
-         qemu-server (>= 6.2-17),
+         qemu-server (>= 7.2-4),
          rsync,
diff -u1 -r pve-old/pve-manager/www/manager6/Utils.js pve-maker/pve-manager/www/manager6/Utils.js
--- pve-old/pve-manager/www/manager6/Utils.js	2022-10-10 08:23:17.646027509 +0000
+++ pve-maker/pve-manager/www/manager6/Utils.js	2022-10-10 08:15:10.615886890 +0000
@@ -598,2 +598,3 @@
 	'virtio-gl': 'VirGL GPU',
+        ramfb: 'ramfb',
 	none: Proxmox.Utils.noneText,
diff -u1 -r pve-old/pve-manager/www/manager6/form/CPUModelSelector.js pve-maker/pve-manager/www/manager6/form/CPUModelSelector.js
--- pve-old/pve-manager/www/manager6/form/CPUModelSelector.js	2022-10-10 08:23:17.662027577 +0000
+++ pve-maker/pve-manager/www/manager6/form/CPUModelSelector.js	2022-10-10 08:07:09.645431881 +0000
@@ -17,3 +17,3 @@
 
-    emptyText: Proxmox.Utils.defaultText + ' (kvm64)',
+    emptyText: Proxmox.Utils.defaultText + ' (host)',
     allowBlank: true,
diff -u1 -r pve-old/pve-manager/www/manager6/qemu/CreateWizard.js pve-maker/pve-manager/www/manager6/qemu/CreateWizard.js
--- pve-old/pve-manager/www/manager6/qemu/CreateWizard.js	2022-10-10 08:23:17.682027661 +0000
+++ pve-maker/pve-manager/www/manager6/qemu/CreateWizard.js	2022-10-10 08:20:31.089317418 +0000
@@ -140,3 +140,3 @@
 		    },
-		    confid: 'ide2',
+		    confid: 'scsi2',
 		    insideWizard: true,
diff -u1 -r pve-old/pve-manager/www/manager6/qemu/MultiHDEdit.js pve-maker/pve-manager/www/manager6/qemu/MultiHDEdit.js
--- pve-old/pve-manager/www/manager6/qemu/MultiHDEdit.js	2022-10-10 08:23:17.686027678 +0000
+++ pve-maker/pve-manager/www/manager6/qemu/MultiHDEdit.js	2022-10-10 08:21:10.769488199 +0000
@@ -40,3 +40,3 @@
 	    return {
-		ide2: 'media=cdrom',
+		scsi2: 'media=cdrom',
 		scsihw: vm.get('current.scsihw'),
diff -u1 -r pve-old/qemu-server/PVE/QemuServer/CPUConfig.pm pve-maker/qemu-server/PVE/QemuServer/CPUConfig.pm
--- pve-old/qemu-server/PVE/QemuServer/CPUConfig.pm	2022-10-10 08:23:05.665976984 +0000
+++ pve-maker/qemu-server/PVE/QemuServer/CPUConfig.pm	2022-10-10 07:59:12.098135536 +0000
@@ -466,3 +466,3 @@
     if ($arch eq 'aarch64') {
-	$cputype = 'cortex-a57';
+	$cputype = 'host';
     }
diff -u1 -r pve-old/qemu-server/PVE/QemuServer.pm pve-maker/qemu-server/PVE/QemuServer.pm
--- pve-old/qemu-server/PVE/QemuServer.pm	2022-10-10 08:24:15.054268607 +0000
+++ pve-maker/qemu-server/PVE/QemuServer.pm	2022-10-10 08:16:47.224329480 +0000
@@ -200,3 +200,3 @@
 	default_key => 1,
-	enum => [qw(cirrus qxl qxl2 qxl3 qxl4 none serial0 serial1 serial2 serial3 std virtio virtio-gl vmware)],
+	enum => [qw(cirrus qxl qxl2 qxl3 qxl4 none serial0 serial1 serial2 serial3 std virtio virtio-gl vmware ramfb)],
     },
@@ -604,3 +604,3 @@
 	optional => 1,
-	type => 'string', format => 'pve-qm-ide',
+	type => 'string', format => 'pve-qm-scsi',
 	typetext => '<volume>',
@@ -1842,2 +1842,3 @@
     'virtio-gl' => 'virtio-vga-gl',
+    'ramfb' => 'ramfb',
 };
@@ -3800,5 +3801,8 @@
     if ($vga->{type} && $vga->{type} !~ m/^serial\d+$/ && $vga->{type} ne 'none'){
-	push @$devices, '-device', print_vga_device(
-	    $conf, $vga, $arch, $machine_version, $machine_type, undef, $qxlnum, $bridges);
-
+	if ($vga->{type} eq 'ramfb'){
+           push @$devices, '-device', 'ramfb';
+        } else {
+           push @$devices, '-device', print_vga_device(
+	   $conf, $vga, $arch, $machine_version, $machine_type, undef, $qxlnum, $bridges);
+    }
 	push @$cmd, '-display', 'egl-headless,gl=core' if $vga->{type} eq 'virtio-gl'; # VIRGL
diff -u1 -r pve-old/qemu-server/debian/changelog pve-maker/qemu-server/debian/changelog
--- pve-old/qemu-server/debian/changelog	2022-10-10 08:24:15.046268574 +0000
+++ pve-maker/qemu-server/debian/changelog	2022-10-10 08:17:47.876601814 +0000
@@ -1 +1,11 @@
+qemu-server (7.2-4) bullseye; urgency=medium
+
+  * add default cpu type to host
+  
+  * add ramfb support
+
+  * set default cdrom to scsi2
+
+ -- foxi <jiangcuo@bingsin.com>  Mon, 10 Oct 2022 16:00:20 +0200
+
 qemu-server (7.2-4) bullseye; urgency=medium  
diff -u1 -r pve-old/qemu-server/debian/control pve-maker/qemu-server/debian/control
--- pve-old/qemu-server/debian/control	2022-10-10 08:24:15.050268590 +0000
+++ pve-maker/qemu-server/debian/control	2022-10-10 08:38:03.203670606 +0000
@@ -45,2 +45,3 @@
          pve-cluster,
+         pve-manager (>= 7.2-11),
          pve-edk2-firmware (>= 3.20210831-1),
