diff -ur pve-src/proxmox-mini-journalreader/src/mini-journalreader.c pve-maker/proxmox-mini-journalreader/src/mini-journalreader.c
--- pve-src/proxmox-mini-journalreader/src/mini-journalreader.c 2022-10-09 14:07:17.107884890 +0000
+++ pve-maker/proxmox-mini-journalreader/src/mini-journalreader.c       2023-01-03 08:08:29.275105219 +0000
@@ -174,7 +174,8 @@
     if (json) {
        char tmp[7];
        for (size_t i = 0; i < l;i++) {
-           if (d[i] == '"' || d[i] == '\\' || (d[i] >= 0 && d[i] <= 0x1F)) {
+               int x = (int)d[i];
+               if (d[i] == '"' || d[i] == '\\' || (x >= 0 && x <= 0x1F)) {
                sprintf(tmp, "\\u%04X", d[i]);
                print_to_buf(tmp, 6);
            } else {
@@ -262,11 +263,11 @@
     const char *endcursor = NULL;
     uint64_t begin = 0;
     uint64_t end = 0;
-    char c;
+    int c;
 
     progname = argv[0];
 
-    while ((c = (char)getopt (argc, argv, "b:e:d:n:f:t:jh")) != -1) {
+while ((c = getopt (argc, argv, "b:e:d:n:f:t:jh")) != -1) {
         switch (c) {
             case 'b':
                 begin = arg_to_uint64(optarg);
