diff -ur pve-src/pve-cluster/data/src/database.c pve-maker/pve-cluster/data/src/database.c
--- pve-src/pve-cluster/data/src/database.c	2022-10-09 14:07:27.623922140 +0000
+++ pve-maker/pve-cluster/data/src/database.c	2023-01-31 11:22:09.644419009 +0000
@@ -351,7 +351,7 @@
 
 		if (te->type == DT_REG) {
 			if (size > 0)
-				te->data.value = g_memdup(value, size);
+				te->data.value = g_memdup2(value, size);
 		} else if (te->type == DT_DIR) {
 			if (size) {
 				cfs_critical("directory inode contains data (inode = %016" PRIX64 ")",
diff -ur pve-src/pve-cluster/data/src/dcdb.c pve-maker/pve-cluster/data/src/dcdb.c
--- pve-src/pve-cluster/data/src/dcdb.c	2022-10-09 14:07:27.623922140 +0000
+++ pve-maker/pve-cluster/data/src/dcdb.c	2023-01-31 11:22:09.648419018 +0000
@@ -347,7 +347,7 @@
 	te->type = type;
 
 	if (te->type == DT_REG && te->size) {
-		te->data.value = g_memdup(data, te->size);
+		te->data.value = g_memdup2(data, te->size);
 		if (!te->data.value) {
 			memdb_tree_entry_free(te);
 			return NULL;
diff -ur pve-src/pve-cluster/data/src/dfsm.c pve-maker/pve-cluster/data/src/dfsm.c
--- pve-src/pve-cluster/data/src/dfsm.c	2022-10-09 14:07:27.623922140 +0000
+++ pve-maker/pve-cluster/data/src/dfsm.c	2023-01-31 11:22:09.648419018 +0000
@@ -460,7 +460,7 @@
 		
 	qm->nodeid = nodeid;
 	qm->pid = pid;
-	qm->msg = g_memdup (msg, msg_len);
+	qm->msg = g_memdup2 (msg, msg_len);
 	qm->msg_len = msg_len;
 	qm->msg_count =  msg_count;
 
@@ -877,7 +877,7 @@
 				goto leave;
 			}
 
-			ni->state = g_memdup(msg, msg_len);
+			ni->state = g_memdup2(msg, msg_len);
 			ni->state_len = msg_len;
 
 			int received_all = 1;
diff -ur pve-src/pve-cluster/data/src/logger.c pve-maker/pve-cluster/data/src/logger.c
--- pve-src/pve-cluster/data/src/logger.c	2022-10-09 14:07:27.623922140 +0000
+++ pve-maker/pve-cluster/data/src/logger.c	2023-01-31 11:22:09.648419018 +0000
@@ -606,7 +606,7 @@
 	}
 
 	*res_len = clog_size(cl->base);
-	gpointer msg = g_memdup(cl->base, *res_len);
+	gpointer msg = g_memdup2(cl->base, *res_len);
 
 	g_mutex_unlock(&cl->mutex);
 
diff -ur pve-src/pve-cluster/data/src/memdb.c pve-maker/pve-cluster/data/src/memdb.c
--- pve-src/pve-cluster/data/src/memdb.c	2022-10-09 14:07:27.623922140 +0000
+++ pve-maker/pve-cluster/data/src/memdb.c	2023-01-31 11:22:09.648419018 +0000
@@ -73,7 +73,7 @@
 	cpy->size = te->size;
 
 	if (with_data && te->size && te->type == DT_REG) {
-		cpy->data.value = g_memdup(te->data.value, te->size);
+		cpy->data.value = g_memdup2(te->data.value, te->size);
 	} else { 
 		cpy->data.value = NULL;
 	}
@@ -676,7 +676,7 @@
 
 	if ((te = memdb_lookup_path(memdb, path, &parent))) {
 		if (te->type == DT_REG) {
-			*data_ret = g_memdup(te->data.value, te->size);
+			*data_ret = g_memdup2(te->data.value, te->size);
 			guint32 size = te->size;
 			return size;
 		}
@@ -1451,7 +1451,7 @@
 		return NULL;
 	}
 
- 	memdb_index_t *copy = (memdb_index_t *)g_memdup(idx, bytes);
+ 	memdb_index_t *copy = (memdb_index_t *)g_memdup2(idx, bytes);
 
 	return copy;
 }
diff -ur pve-src/pve-cluster/data/src/status.c pve-maker/pve-cluster/data/src/status.c
--- pve-src/pve-cluster/data/src/status.c	2022-12-07 01:38:10.135624260 +0000
+++ pve-maker/pve-cluster/data/src/status.c	2023-01-31 11:22:09.652419026 +0000
@@ -1101,14 +1101,14 @@
 		g_hash_table_remove(kvhash, key);
 	} else if ((entry = (kventry_t *)g_hash_table_lookup(kvhash, key))) {
 		g_free(entry->data);
-		entry->data = g_memdup(data, len);
+		entry->data = g_memdup2(data, len);
 		entry->len = len;
 		entry->version++;
 	} else {
 		kventry_t *entry = g_new0(kventry_t, 1);
 
 		entry->key = g_strdup(key);
-		entry->data = g_memdup(data, len);
+		entry->data = g_memdup2(data, len);
 		entry->len = len;
 
 		g_hash_table_replace(kvhash, entry->key, entry);
@@ -1465,14 +1465,14 @@
 	rrdentry_t *entry;
 	if ((entry = (rrdentry_t *)g_hash_table_lookup(rrdhash, key))) {
 		g_free(entry->data);
-		entry->data = g_memdup(data, len);
+		entry->data = g_memdup2(data, len);
 		entry->len = len;
 		entry->time = time(NULL);
 	} else {
 		rrdentry_t *entry = g_new0(rrdentry_t, 1);
 
 		entry->key = g_strdup(key);
-		entry->data = g_memdup(data, len);
+		entry->data = g_memdup2(data, len);
 		entry->len = len;
 		entry->time = time(NULL);
