diff -ur -x .git pve-src/pve-installer/proxinstall pve-maker/pve-installer/proxinstall
--- pve-src/pve-installer/proxinstall	2023-03-23 10:25:19.038996029 +0000
+++ pve-maker/pve-installer/proxinstall	2023-03-23 11:05:35.497859903 +0000
@@ -3683,15 +3683,15 @@
 	    "See 'System Requirements' in the $setup->{fullname} documentation.");
     }
 
-    if ($setup->{product} eq 'pve') {
-	eval {
-	    my $cpuinfo = file_get_contents('/proc/cpuinfo');
-	    if ($cpuinfo && !($cpuinfo =~ /^flags\s*:.*(vmx|svm)/m)) {
-		display_error("No support for KVM virtualization detected.\n\n" .
-			      "Check BIOS settings for Intel VT / AMD-V / SVM.")
-	    }
-	};
-    }
+#    if ($setup->{product} eq 'pve') {
+#	eval {
+#	    my $cpuinfo = file_get_contents('/proc/cpuinfo');
+#	    if ($cpuinfo && !($cpuinfo =~ /^flags\s*:.*(vmx|svm)/m)) {
+#		display_error("No support for KVM virtualization detected.\n\n" .
+#			      "Check BIOS settings for Intel VT / AMD-V / SVM.")
+#	    }
+#	};
+#    }
 
     display_html();
 
