diff -ur -x .git pve-src/pve-manager/configs/Makefile pve-maker/pve-manager/configs/Makefile
--- pve-src/pve-manager/configs/Makefile	2023-03-30 14:54:49.541560031 +0000
+++ pve-maker/pve-manager/configs/Makefile	2023-03-30 14:53:47.809003936 +0000
@@ -8,7 +8,7 @@
 .PHONY: install
 install: country.dat vzdump.conf pve-sources.list pve-initramfs.conf pve-blacklist.conf pve.logrotate
 	install -D -m 0644 pve.logrotate ${DESTDIR}/etc/logrotate.d/pve
-	install -D -m 0644 pve-sources.list ${DESTDIR}/etc/apt/sources.list.d/pve-enterprise.list
+	install -D -m 0644 pve-sources.list ${DESTDIR}/etc/apt/sources.list.d/pve-arm64.list
 	install -D -m 0644 pve-blacklist.conf ${DESTDIR}/etc/modprobe.d/pve-blacklist.conf
 	install -D -m 0644 vzdump.conf ${DESTDIR}/etc/vzdump.conf
 	install -D -m 0644 pve-initramfs.conf ${DESTDIR}/etc/initramfs-tools/conf.d/pve-initramfs.conf
diff -ur -x .git pve-src/pve-manager/configs/pve-sources.list pve-maker/pve-manager/configs/pve-sources.list
--- pve-src/pve-manager/configs/pve-sources.list	2023-03-30 14:54:49.541560031 +0000
+++ pve-maker/pve-manager/configs/pve-sources.list	2023-03-30 14:52:57.692555512 +0000
@@ -1 +1 @@
-deb https://enterprise.proxmox.com/debian/pve bullseye pve-enterprise
+deb https://mirrors.apqa.cn/proxmox/ pvearm main
diff -ur -x .git pve-src/pve-manager/debian/lintian-overrides pve-maker/pve-manager/debian/lintian-overrides
--- pve-src/pve-manager/debian/lintian-overrides	2023-03-30 14:54:49.545560067 +0000
+++ pve-maker/pve-manager/debian/lintian-overrides	2023-03-30 15:06:14.163900101 +0000
@@ -2,6 +2,6 @@
 pve-manager: no-manual-page usr/bin/pvebanner
 pve-manager: no-manual-page usr/bin/pveupdate
 pve-manager: non-standard-dir-perm var/log/pveproxy/ 0700 != 0755
-pve-manager: package-installs-apt-sources etc/apt/sources.list.d/pve-enterprise.list
+pve-manager: package-installs-apt-sources etc/apt/sources.list.d/pve-arm64.list
 pve-manager: privacy-breach-generic usr/share/pve-manager/touch/sencha-touch-all-debug.js *
 pve-manager: systemd-service-file-refers-to-unusual-wantedby-target lib/systemd/system/pvebanner.service getty.target
