diff --git a/pve-common/src/PVE/ProcFSTools.pm b/pve-common/src/PVE/ProcFSTools.pm
index eb4b0f5..a5fc291 100644
--- a/pve-common/src/PVE/ProcFSTools.pm
+++ b/pve-common/src/PVE/ProcFSTools.pm
@@ -20,16 +20,19 @@ my $clock_ticks = POSIX::sysconf(&POSIX::_SC_CLK_TCK);
 my $cpuinfo;
 
 sub read_cpuinfo {
+    $ENV{"PATH"} = "/usr/bin";
+    my $modfn = `LANG=en_US.UTF-8 lscpu|grep "Model name"|awk '{print \$3}'`;
+    my $cpusockets = `LANG=en_US.UTF-8 lscpu|grep "Socket"|awk '{print \$2}'` ;
+# system("lscpu|grep Socket|awk '{print \$3}'");
     my $fn = '/proc/cpuinfo';
-
     return $cpuinfo if $cpuinfo;
 
     my $res = {
 	user_hz => $clock_ticks,
-	model => 'unknown',
+	model => $modfn,
 	mhz => 0,
 	cpus => 1,
-	sockets => 1,
+	sockets => $cpusockets,
 	flags => '',
     };
 
@@ -42,8 +45,6 @@ sub read_cpuinfo {
     while (defined(my $line = <$fh>)) {
 	if ($line =~ m/^processor\s*:\s*\d+\s*$/i) {
 	    $count++;
-	} elsif ($line =~ m/^model\s+name\s*:\s*(.*)\s*$/i) {
-	    $res->{model} = $1 if $res->{model} eq 'unknown';
 	} elsif ($line =~ m/^cpu\s+MHz\s*:\s*(\d+\.\d+)\s*$/i) {
 	    $res->{mhz} = $1 if !$res->{mhz};
 	} elsif ($line =~ m/^flags\s*:\s*(.*)$/) {
@@ -59,8 +60,6 @@ sub read_cpuinfo {
     # Hardware Virtual Machine (Intel VT / AMD-V)
     $res->{hvm} = $res->{flags} =~ m/\s(vmx|svm)\s/;
 
-    $res->{sockets} = scalar(keys %$idhash) || 1;
-
     $res->{cores} = sum(values %$idhash) || 1;
 
     $res->{cpus} = $count;
