Subject: [PATCH] Fixed the issue of ssh-key verification error during lxc creation
From: peio42
Date: 2023-04-15
References: https://github.com/jiangcuo/Proxmox-Arm64/issues/16
diff -x .git -ur pve-src/pve-common/src/PVE/Tools.pm pve-maker/pve-common/src/PVE/Tools.pm
--- pve-src/pve-common/src/PVE/Tools.pm 2023-04-01 09:33:45.500762146 +0800
+++ pve-maker/pve-common/src/PVE/Tools.pm       2023-04-15 00:05:26.189684756 +0800
@@ -100,7 +100,7 @@
 
 use constant {O_PATH    => 0x00200000,
               O_CLOEXEC => 0x00080000,
-              O_TMPFILE => 0x00410000}; # This includes O_DIRECTORY
+              O_TMPFILE => 0x00400000 | O_DIRECTORY};
 
 use constant {AT_EMPTY_PATH => 0x1000,
               AT_FDCWD => -100};
