diff -uNr -x .git a/qemu-server/PVE/QemuServer.pm b/qemu-server/PVE/QemuServer.pm
--- a/qemu-server/PVE/QemuServer.pm	2023-04-27 06:25:27.919683004 -0400
+++ b/qemu-server/PVE/QemuServer.pm	2023-04-27 06:52:27.294886400 -0400
@@ -89,6 +89,22 @@ my $OVMF = {
        ],
     },
     aarch64 => {
+    '4m-no-smm' => [
+           "$EDK2_FW_BASE/AAVMF_CODE.fd",
+           "$EDK2_FW_BASE/AAVMF_VARS.fd",
+       ],
+       '4m-no-smm-ms' => [
+           "$EDK2_FW_BASE/AAVMF_CODE.ms.fd",
+           "$EDK2_FW_BASE/AAVMF_VARS.ms.fd",
+       ],
+       '4m' => [
+           "$EDK2_FW_BASE/AAVMF_CODE.fd",
+           "$EDK2_FW_BASE/AAVMF_VARS.fd",
+       ],
+       '4m-ms' => [
+           "$EDK2_FW_BASE/AAVMF_CODE.ms.fd",
+           "$EDK2_FW_BASE/AAVMF_VARS.ms.fd",
+       ],
        default => [
            "$EDK2_FW_BASE/AAVMF_CODE.fd",
            "$EDK2_FW_BASE/AAVMF_VARS.fd",
@@ -3397,7 +3420,7 @@ sub get_ovmf_files($$$) {
        or die "no OVMF images known for architecture '$arch'\n";
 
     my $type = 'default';
-    if ($arch ne "aarch64" && defined($efidisk->{efitype}) && $efidisk->{efitype} eq '4m') {
+    if ( defined($efidisk->{efitype}) && $efidisk->{efitype} eq '4m') {
        $type = $smm ? "4m" : "4m-no-smm";
        $type .= '-ms' if $efidisk->{'pre-enrolled-keys'};
     }
