diff -x .git -ur a/PVE/QemuServer/USB.pm b/PVE/QemuServer/USB.pm
--- a/PVE/QemuServer/USB.pm	2023-06-06 13:30:40.391616859 +0000
+++ b/PVE/QemuServer/USB.pm	2023-06-06 13:28:43.439549799 +0000
@@ -59,8 +59,8 @@
     my $is_q35 = PVE::QemuServer::Machine::machine_type_is_q35($conf);
 
     if ($arch eq 'aarch64') {
-        $pciaddr = print_pci_addr('ehci', $bridges, $arch, $machine);
-        push @$devices, '-device', "usb-ehci,id=ehci$pciaddr";
+        $pciaddr = print_pci_addr('qemu-xhci', $bridges, $arch, $machine);
+        push @$devices, '-device', "qemu-xhci,id=qemu-xhci";
     } elsif (!$is_q35) {
         $pciaddr = print_pci_addr("piix3", $bridges, $arch, $machine);
         push @$devices, '-device', "piix3-usb-uhci,id=uhci$pciaddr.0x2";
diff -x .git -ur a/PVE/QemuServer.pm b/PVE/QemuServer.pm
--- a/PVE/QemuServer.pm	2023-06-06 12:57:18.378594124 +0000
+++ b/PVE/QemuServer.pm	2023-06-06 13:29:58.967574871 +0000
@@ -1473,7 +1473,7 @@
     # we use uhci for old VMs because tablet driver was buggy in older qemu
     my $usbbus;
     if ($q35 || $arch eq 'aarch64') {
-	$usbbus = 'ehci';
+	$usbbus = 'qemu-xhci';
     } else {
 	$usbbus = 'uhci';
     }
@@ -1486,7 +1486,7 @@
 
     return if $arch ne 'aarch64';
 
-    return "usb-kbd,id=keyboard,bus=ehci.0,port=2";
+    return "usb-kbd,id=keyboard,bus=qemu-xhci.0,port=2";
 }
 
 my sub get_drive_id {
