diff -ur -x .git a/proxmox-backup-qemu/Cargo.toml b/proxmox-backup-qemu/Cargo.toml
--- a/proxmox-backup-qemu/Cargo.toml        2022-09-12 00:52:28.477933067 -0400
+++ b/proxmox-backup-qemu/Cargo.toml        2022-09-12 00:51:40.433801344 -0400
@@ -34,13 +34,36 @@
 proxmox-schema = { version = "1", features = [ "api-macro" ] }
 proxmox-sys = { version = "0.2", features = [ "sortable-macro"] }
 
-pbs-api-types = { git = "git://git.proxmox.com/git/proxmox-backup.git", rev = "bc21ade293fa7af096ad21fce54fda1e9814993c" }
-pbs-tools     = { git = "git://git.proxmox.com/git/proxmox-backup.git", rev = "bc21ade293fa7af096ad21fce54fda1e9814993c" }
-pbs-config    = { git = "git://git.proxmox.com/git/proxmox-backup.git", rev = "bc21ade293fa7af096ad21fce54fda1e9814993c" }
-pbs-datastore = { git = "git://git.proxmox.com/git/proxmox-backup.git", rev = "bc21ade293fa7af096ad21fce54fda1e9814993c" }
-pbs-client    = { git = "git://git.proxmox.com/git/proxmox-backup.git", rev = "bc21ade293fa7af096ad21fce54fda1e9814993c" }
+pbs-api-types = "0.1.0"
+pbs-tools     = "0.1.0"
+pbs-config    = "0.1.0"
+pbs-datastore = "0.1.0"
+pbs-client    = "0.1.0"
 
 serde_json = "1.0"
 tokio = { version = "1.6", features = [ "fs", "io-util", "macros", "net", "rt-multi-thread", "signal", "time" ] }
 tokio-stream = "0.1.1"
 bincode = "1.0"
+
+[patch.crates-io]
+proxmox-async = { path = "../proxmox/proxmox-async" }
+proxmox-lang = { path = "../proxmox/proxmox-lang" }
+proxmox-schema = { path = "../proxmox/proxmox-schema" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
+proxmox-serde = { path = "../proxmox/proxmox-serde" }
+proxmox-time = { path = "../proxmox/proxmox-time" }
+proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
+proxmox-compression = { path = "../proxmox/proxmox-compression" }
+proxmox-fuse = { path = "../proxmox-fuse" }
+proxmox-http = { path = "../proxmox/proxmox-http" }
+proxmox-io = { path = "../proxmox/proxmox-io" }
+proxmox-router = { path = "../proxmox/proxmox-router" }
+pxar = { path = "../pxar" }
+proxmox-section-config = { path = "../proxmox/proxmox-section-config" }
+proxmox-shared-memory = { path = "../proxmox/proxmox-shared-memory" }
+proxmox-borrow = { path = "../proxmox/proxmox-borrow" }
+pbs-api-types= { path = "../proxmox-backup/pbs-api-types" }
+pbs-tools= { path = "../proxmox-backup/pbs-tools" }
+pbs-config= { path = "../proxmox-backup/pbs-config" }
+pbs-datastore= { path = "../proxmox-backup/pbs-datastore" }
+pbs-client= { path = "../proxmox-backup/pbs-client" }
diff -ur -x .git a/proxmox-backup-qemu/debian/control b/proxmox-backup-qemu/debian/control
--- a/proxmox-backup-qemu/debian/control    2022-09-12 00:52:28.477933067 -0400
+++ b/proxmox-backup-qemu/debian/control    2022-09-12 00:51:40.433801344 -0400
@@ -4,83 +4,6 @@
 Maintainer: Proxmox Support Team <support@proxmox.com>
 Build-Depends: cbindgen,
  debhelper (>= 12),
- librust-anyhow-1+default-dev,
- librust-base64-0.13+default-dev,
- librust-bincode-1+default-dev,
- librust-bitflags-1+default-dev (>= 1.2.1-~~),
- librust-bytes-1+default-dev,
- librust-crc32fast-1+default-dev,
- librust-endian-trait-0.6+arrays-dev,
- librust-endian-trait-0.6+default-dev,
- librust-flate2-1+default-dev,
- librust-foreign-types-0.3+default-dev,
- librust-futures-0.3+default-dev,
- librust-h2-0.3+default-dev,
- librust-h2-0.3+stream-dev,
- librust-hex-0.4+default-dev (>= 0.4.3-~~),
- librust-hex-0.4+serde-dev (>= 0.4.3-~~),
- librust-http-0.2+default-dev,
- librust-hyper-0.14+default-dev,
- librust-hyper-0.14+full-dev,
- librust-lazy-static-1+default-dev (>= 1.4-~~),
- librust-libc-0.2+default-dev,
- librust-log-0.4+default-dev,
- librust-nix-0.19+default-dev (>= 0.19.1-~~),
- librust-nom-5+default-dev (>= 5.1-~~),
- librust-once-cell-1+default-dev (>= 1.5-~~),
- librust-openssl-0.10+default-dev,
- librust-pathpatterns-0.1+default-dev (>= 0.1.2-~~),
- librust-percent-encoding-2+default-dev (>= 2.1-~~),
- librust-pin-project-lite-0.2+default-dev,
- librust-proxmox-async-0.4+default-dev,
- librust-proxmox-borrow-1+default-dev,
- librust-proxmox-compression-0.1+default-dev (>= 0.1.1-~~),
- librust-proxmox-fuse-0.1+default-dev (>= 0.1.1-~~),
- librust-proxmox-http-0.6+client-dev,
- librust-proxmox-http-0.6+default-dev,
- librust-proxmox-http-0.6+http-helpers-dev,
- librust-proxmox-http-0.6+websocket-dev,
- librust-proxmox-io-1+default-dev (>= 1.0.1-~~),
- librust-proxmox-io-1+tokio-dev (>= 1.0.1-~~),
- librust-proxmox-lang-1+default-dev (>= 1.1-~~),
- librust-proxmox-router-1+cli-dev (>= 1.2-~~),
- librust-proxmox-router-1+default-dev (>= 1.2-~~),
- librust-proxmox-schema-1+api-macro-dev (>= 1.3.1-~~),
- librust-proxmox-schema-1+default-dev (>= 1.3.1-~~),
- librust-proxmox-section-config-1+default-dev,
- librust-proxmox-serde-0.1+default-dev,
- librust-proxmox-shared-memory-0.2+default-dev,
- librust-proxmox-sys-0.2+default-dev,
- librust-proxmox-sys-0.2+sortable-macro-dev,
- librust-proxmox-time-1+default-dev (>= 1.1.1-~~),
- librust-proxmox-uuid-1+default-dev,
- librust-proxmox-uuid-1+serde-dev,
- librust-pxar-0.10+default-dev (>= 0.10.1-~~),
- librust-pxar-0.10+tokio-io-dev (>= 0.10.1-~~),
- librust-regex-1+default-dev (>= 1.5.5-~~),
- librust-rustyline-7+default-dev,
- librust-serde-1+default-dev,
- librust-serde-1+derive-dev,
- librust-serde-json-1+default-dev,
- librust-serde-plain-1+default-dev,
- librust-tar-0.4+default-dev,
- librust-tokio-1+default-dev (>= 1.6-~~),
- librust-tokio-1+fs-dev (>= 1.6-~~),
- librust-tokio-1+io-util-dev (>= 1.6-~~),
- librust-tokio-1+macros-dev (>= 1.6-~~),
- librust-tokio-1+net-dev (>= 1.6-~~),
- librust-tokio-1+rt-dev (>= 1.6-~~),
- librust-tokio-1+rt-multi-thread-dev (>= 1.6-~~),
- librust-tokio-1+signal-dev (>= 1.6-~~),
- librust-tokio-1+sync-dev (>= 1.6-~~),
- librust-tokio-1+time-dev (>= 1.6-~~),
- librust-tokio-stream-0.1+default-dev (>= 0.1.1-~~),
- librust-tower-service-0.3+default-dev,
- librust-url-2+default-dev (>= 2.1-~~),
- librust-walkdir-2+default-dev,
- librust-xdg-2+default-dev (>= 2.2-~~),
- librust-zstd-0.6+bindgen-dev,
- librust-zstd-0.6+default-dev,
 Standards-Version: 4.5.1
 Homepage: https://www.proxmox.com
